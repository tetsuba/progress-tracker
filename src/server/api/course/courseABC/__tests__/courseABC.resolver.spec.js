const { createTestServer } = require('../../../../test/serverTestHelper')
const {
  CourseABCMockQuerySuccess,
  CourseABCMockMutationSuccess,
} = require('../../../../../test/mockApi/course/courseABC/courseABCMockData')
const {
  StudentFirstNameMockQuerySuccess,
} = require('../../../../../test/mockApi/student/studentMockData')
const {
  GET_STUDENT_COURSE_ABC,
} = require('../../../../../client/api/course/courseABC/courseABC.query')
const {
  SAVE_COURSE_PROGRESS_MUTATION,
} = require('../../../../../client/api/course/courseABC/courseABC.mutation')

describe('courseABC.resolver', () => {
  describe('@Query', () => {
    describe('getStudentCourseABC', () => {
      const ctx = {
        models: {
          Student: new StudentFirstNameMockQuerySuccess(),
          CourseABC: new CourseABCMockQuerySuccess(),
        },
      }

      test('should return course history and student name', async () => {
        const { query } = createTestServer(ctx)
        const res = await query({
          query: GET_STUDENT_COURSE_ABC,
          variables: { input: { id: 'student1' } },
        })
        expect(res).toMatchSnapshot()
      })
    })
  })
  describe('@Mutation', () => {
    describe('saveCourseProgress', () => {
      const ctx = {
        models: {
          Student: new StudentFirstNameMockQuerySuccess(),
          CourseABC: new CourseABCMockMutationSuccess(),
        },
      }

      test('should return course history and student name', async () => {
        const alphabet = [
          { letter: 'A', value: 1 },
          { letter: 'B', value: 1 },
          { letter: 'C', value: 1 },
        ]
        const date = 'today is the date'
        const { mutate } = createTestServer(ctx)
        const res = await mutate({
          mutation: SAVE_COURSE_PROGRESS_MUTATION,
          variables: {
            input: {
              id: 'student1',
              date: date,
              alphabet: alphabet,
            },
          },
        })
        expect(res).toMatchSnapshot()
      })
    })
  })
})
