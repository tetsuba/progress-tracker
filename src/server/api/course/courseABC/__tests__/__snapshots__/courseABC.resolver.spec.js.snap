// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`courseABC.resolver @Mutation saveCourseProgress should return course history and student name 1`] = `
Object {
  "data": Object {
    "saveCourseProgress": Object {
      "history": Array [
        Object {
          "alphabet": Array [
            Object {
              "letter": "A",
              "value": 0,
            },
            Object {
              "letter": "B",
              "value": 0,
            },
            Object {
              "letter": "C",
              "value": 0,
            },
          ],
          "date": "today",
        },
        Object {
          "alphabet": Array [
            Object {
              "letter": "A",
              "value": 0,
            },
            Object {
              "letter": "B",
              "value": 0,
            },
            Object {
              "letter": "C",
              "value": 0,
            },
          ],
          "date": "tomorrow",
        },
        Object {
          "alphabet": Array [
            Object {
              "letter": "A",
              "value": 1,
            },
            Object {
              "letter": "B",
              "value": 1,
            },
            Object {
              "letter": "C",
              "value": 1,
            },
          ],
          "date": "today is the date",
        },
      ],
      "studentName": "Bob",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`courseABC.resolver @Query getStudentCourseABC should return course history and student name 1`] = `
Object {
  "data": Object {
    "getStudentCourseABC": Object {
      "history": Array [
        Object {
          "alphabet": Array [
            Object {
              "letter": "A",
              "value": 0,
            },
            Object {
              "letter": "B",
              "value": 0,
            },
            Object {
              "letter": "C",
              "value": 0,
            },
          ],
          "date": "today",
        },
        Object {
          "alphabet": Array [
            Object {
              "letter": "A",
              "value": 0,
            },
            Object {
              "letter": "B",
              "value": 0,
            },
            Object {
              "letter": "C",
              "value": 0,
            },
          ],
          "date": "tomorrow",
        },
      ],
      "studentName": "Bob",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
